helper-cli/src/main/kotlin/commands/DownloadResultsFromPostgresCommand.kt

```diff
@@ -0,0 +1,189 @@
+        .default(ortConfigDirectory.resolve(ORT_CONFIG_FILENAME))
+
+    private val configArguments by option(
+        "-P",
+        help = "Override a key-value pair in the configuration file. For example: " +
+                "-P ort.scanner.storages.postgres.schema=testSchema"
+    ).associate()
+
+    private val startId by option(
+        "--start-id",
+        help = "The smallest row-id to download."
+    ).convert { it.toInt() }
+        .default(-1)
+
+    override fun run() {
+        if (!outputDir.exists()) outputDir.mkdirs()
+
+        val localStorage = OrtResultStorage(outputDir)
+        val localRowIds = localStorage.getIds()
+
+        openDatabaseConnection().use { connection ->
+            val remoteRowIds = fetchRowIds(connection)
+            val missingRowIds = remoteRowIds - localRowIds
+            println("Requested the download of ${remoteRowIds.size} of which ${missingRowIds.size} are missing.")
+
+            missingRowIds.forEachIndexed { i, rowId ->
+                println("[${i + 1} / ${missingRowIds.size}] downloading row $rowId")
+
+                val duration = measureTime {
+                    val ortResultJson = fetchOrtResult(connection, rowId)
+
+                    if (ortResultJson != null) localStorage.add(rowId, ortResultJson)
+                }
+
+                println("[${i + 1} / ${missingRowIds.size}] took ${duration.toString(DurationUnit.SECONDS)}s.")
+            }
+        }
+    }
+
+    private fun fetchOrtResult(connection: Connection, rowId: Int): String? {
+        // TODO: The transmission is relatively slow currently due to lack of compression. Check whether
+        // the gzip extension can help speed this up https://github.com/pramsey/pgsql-gzip.
+
+        val query = "SELECT t.$columnName::TEXT AS result FROM $tableName t WHERE id = ?;"
+
+        val resultSet = connection.prepareStatement(query).apply { setInt(1, rowId) }.executeQuery()
+
+        return if (resultSet.next()) {
+            resultSet.getString("result")
+        } else {
+            null
+        }
+    }
+
+    private fun fetchRowIds(connection: Connection): Set<Int> {
+        val query = "SELECT t.id FROM $tableName t WHERE id >= ?;"
+
+        val resultSet = connection.prepareStatement(query).apply {
+            setInt(1, startId)
+        }.executeQuery()
+
+        val result = mutableSetOf<Int>()
+
+        while (resultSet.next()) {
+            result += resultSet.getInt("id")
+        }
+
+        return result
+    }
+
+    private fun openDatabaseConnection(): Connection {
+        val storageConfig = OrtConfiguration.load(configArguments, configFile).scanner.storages?.values
+            ?.filterIsInstance<PostgresStorageConfiguration>()?.firstOrNull()
+            ?: throw IllegalArgumentException("postgresStorage not configured.")
+
+        return DatabaseUtils.createHikariDataSource(
+            config = storageConfig,
+            applicationNameSuffix = ORTH_NAME,
```

Comment:
You could make a separate commit and align this in the upload command.

Reply:
The upload command is a sub command of `ort`, while this one is a sub command `orth`.